src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -82,4 +80,73 @@ public void updateMeal(int mealId, Meal meal) {
         System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
     }
 
-}
\ No newline at end of file
+
+    @Override
+    public Map<String, List<Meal>> mealsByCategories() {
+        List<Meal> allMeals = getMeals();
+        Map<String, List<Meal>> mealsByCategoriesList = new HashMap<>();
+        for (Meal meal : allMeals) {
+            mealsByCategoriesList.computeIfAbsent(meal.getCategory(), k -> new ArrayList<>()).add(meal);
+        }
+
+        return mealsByCategoriesList;
+    }
+
+    private int randomNumberGenerator(int max) {
+        Random random = new Random();
+        return random.nextInt(max);
+    }
+
+
+    @Override
+    // generate 7 random meals per each day
+    public HashMap<DaysOfTheWeek, Meal> generateWeeklyMeals() {
+
+        HashMap<DaysOfTheWeek, Meal> weeklyPlan = new HashMap<>();
+        Map<String, List<Meal>> mealsByCategories = mealsByCategories();
+
+        // create a method ↓
+        HashMap<String, Integer> request = new HashMap<>();
+        request.put("Chicken", 2);
+        request.put("Fish", 1);
+        request.put("Meat", 2);
+        request.put("Vegetarian", 2);
+
+
+        for (DaysOfTheWeek day : DaysOfTheWeek.values()) {
+            String randomCategory = (String) request.keySet().toArray()[new Random().nextInt(request.keySet().toArray().length)];
+            int requestNumber = request.get(randomCategory);
+
+            while (requestNumber == 0) {
+                randomCategory = (String) request.keySet().toArray()[new Random().nextInt(request.keySet().toArray().length)];
+                requestNumber = request.get(randomCategory);
+            }
+
+            if ((requestNumber > 0)) {
```

Comment:
If the category is removed, no need to have this check, as the requestNumber will always be greater than 0